extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -212,6 +204,97 @@ export class InstrumentationService {
   }
 
   /**
+   * Creates listeners for visibility conditions or calls the callback if all
+   * the conditions are met.
+   * @param {!AnalyticsEventListenerDef} The callback to call when the event
+   *   occurs.
+   * @param {!JSONObject} config Configuration for instrumentation.
+   * @private
+   */
+  createVisibilityListener_(callback, config) {
+    if (!this.isVisibilitySpecValid_(config)) {
+      return;
+    }
+
+    // TODO(avimehta, #1297): Add all the listeners so that visibility
+    // conditions are monitored and callback is called when the conditions
+    // are met.
+    if (this.viewer_.isVisible()) {
+      callback(new AnalyticsEvent(AnalyticsEventType.VISIBLE));
+    } else {
+      this.viewer_.onVisibilityChanged(() => {
+        if (this.viewer_.isVisible()) {
+          callback(new AnalyticsEvent(AnalyticsEventType.VISIBLE));
+        }
+      });
+    }
+  }
+
+  /**
+   * Checks and outputs information about visibilitySpecValidation.
+   * @param {!JSONObject} config Configuration for instrumentation.
+   */
+  isVisibilitySpecValid_(config) {
+    if (!config['visibilitySpec'] || !this.isViewabilityExperimentOn_()) {
+      return true;
+    }
+
+    const spec = config['visibilitySpec'];
+    if (!spec['selector'] || spec['selector'][0] != '#') {
+      user.error(this.TAG_, 'Visibility spec requires an id selector');
+      return false;
+    }
+
+    const ctMax = spec['continuousTimeMax'];
+    const ctMin = spec['continuousTimeMin'];
+    const ttMax = spec['totalTimeMax'];
+    const ttMin = spec['totalTimeMin'];
+
+    // "", 1, 0, undefined, 100, 101 are positive. -1, NaN are not.
+    function isPositiveNumber(num) {
+      return num === undefined || Math.sign(num) >= 0;
+    }
+
+    if (!isPositiveNumber(ctMin) || !isPositiveNumber(ctMax) ||
+        !isPositiveNumber(ttMin) || !isPositiveNumber(ttMax)) {
+      user.error(this.TAG_, 'Timing conditions should be positive integers ' +
+          'when specified.');
+      return false;
+    }
+
+    if ((ctMax || ttMax) && !spec['unload']) {
```

Comment:
max conditions dictate the maximum amount of time for this the element is in view. The example you mention is incomplete (or is trivial) because there is no min specified. 

If I just specify a max, I am basically saying "fire this trigger if the amount of time is less than this max". Which technically happens as soon as the item comes into view.

A sensible example for max condition would be: "fire this trigger if the amount of time is less than max at unload". 

Not sure if this makes things clear. I think I might understand your question better if you could give me an example where max is used without unload but still makes sense. (note that I only verify for time max not for amount of element on screen max)